app/Helpers/database/user-auth.php

```diff
@@ -82,7 +82,7 @@ function authenticateForConnect(?string $username, ?string $pass = null, ?string
 
     return [
         'Success' => true,
-        'User' => $user->User,
+        'User' => $user->display_name,
```

Comment:
This should still return `User`. The display name should be returned in `DisplayName`. The client is already ready for that change: https://github.com/RetroAchievements/rcheevos/pull/191

Passing `DisplayName` as `User` causes the avatar to resolve incorrectly
![image](https://github.com/user-attachments/assets/9f327bb7-15f6-4c66-a29d-7cea4de4ac40)

And the request for the patchdata to fail:
```
{"Success":false,"Error":"Invalid user\/token combination.","Code":"invalid_credentials","Status":401}
```

Because `authenticateFromAppToken` is only checking for `User`.

Reply:
Added `DisplayName` return in latest.

Per https://github.com/RetroAchievements/RAWeb/pull/3017#discussion_r1909629207, this does seem to imply that clients are always anticipating `/UserPic/{$username}.png` as the avatar URL.